src/hotspot/cpu/x86/x86.ad

```diff
@@ -8407,6 +8408,57 @@ instruct vmaskcast(vec dst) %{
   ins_pipe(empty);
 %}
 
+instruct vmaskcast_avx(vec dst, vec src) %{
+  predicate(Matcher::vector_length(n) == Matcher::vector_length(n->in(1)) &&
+            Matcher::vector_length_in_bytes(n) != Matcher::vector_length_in_bytes(n->in(1)) &&
+            (UseAVX >= 2 ||
+             (Matcher::vector_length_in_bytes(n) != 32 &&
+              Matcher::vector_length_in_bytes(n->in(1)) != 32)));
+  match(Set dst (VectorMaskCast src));
+  format %{ "vector_mask_cast $dst, $src" %}
+  ins_encode %{
+    int vlen = Matcher::vector_length(this);
+    BasicType src_bt = Matcher::vector_element_basic_type(this, $src);
+    BasicType dst_bt = Matcher::vector_element_basic_type(this);
+    __ vector_mask_cast($dst$$XMMRegister, $src$$XMMRegister, dst_bt, src_bt, vlen);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct vmaskcast_avx1_32B_expand(vec dst, vec src, vec xtmp1, vec xtmp2) %{
+  predicate(UseAVX == 1 && Matcher::vector_length_in_bytes(n) == 32 &&
```

Comment:
Limitation related to VectorLoadMask https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/x86/x86.ad#L1852 will prevent exercising this pattern.

Reply:
Thanks for pointing out this limitation! So the original `VectorCast` pattern also has the similar issue, right? Further thinking about this limitation, since almost all the mask generation/using API will check the `VectorLoadMask`, this rule may not be useful. But keeping it is fine to me in case `VectorMaskCast` is used for other cases or the limitation is relaxed in future. WDYT?

Reply:
Correct it's an existing limitation, but in general any newly added code should be reachable in some execution path of a valid testcase. WDYT ?

Reply:
Make sense to me! So we need to add the same limitation to `VectorMaskCast`?